### PR TITLE
Applies malloc flags across all CBMC proofs

### PR DESCRIPTION
**Pull request by tegansb**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: .cbmc-batch/jobs/aws_cryptosdk_aes_gcm_decrypt/cbmc-batch.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-jobos: ubuntu16
-cbmcflags: "--bounds-check;--div-by-zero-check;--float-overflow-check;--nan-check;--pointer-check;--pointer-overflow-check;--signed-overflow-check;--undefined-shift-check;--unsigned-overflow-check;--unwind;1;--unwinding-assertions;--unwindset;flush_openssl_errors.0:1;--object-bits;8"
-goto: aws_cryptosdk_aes_gcm_decrypt_harness.goto
+cbmcflags: "--flush;--bounds-check;--conversion-check;--div-by-zero-check;--float-overflow-check;--nan-check;--pointer-check;--pointer-overflow-check;--pointer-primitive-check;--signed-overflow-check;--undefined-shift-check;--unsigned-overflow-check;--unwinding-assertions;--malloc-may-fail;--malloc-fail-null;--unwind;1;--unwindset;flush_openssl_errors.0:1;--object-bits;8"
```
[the remark this answers is not in the document]
Comment (feliperodri) [reply .cbmc-batch/jobs/aws_cryptosdk_aes_gcm_decrypt/cbmc-batch.yaml:1]: Perhaps you can close #615, run `prepare.py` locally in this branch, and push all updated `cbmc-batch.yaml` files.

File: .cbmc-batch/jobs/aws_cryptosdk_string_dup/aws_cryptosdk_string_dup_harness.c [hunk 2]
```diff
@@ -23,7 +23,10 @@
 void aws_cryptosdk_string_dup_harness() {
     /* data structure */
     struct aws_allocator *alloc = can_fail_allocator(); /* Precondition: alloc must be non-null */
-    struct aws_string *str_a    = ensure_string_is_allocated_bounded_length(MAX_STRING_LEN);
+    struct aws_string *str_a    = ensure_string_is_allocated_nondet_length();
+
+    __CPROVER_assume(aws_string_is_valid(str_a));
+    __CPROVER_assume(str_a->len <= MAX_STRING_LEN);
```
[the remark this answers is not in the document]
Comment (tegansb) [reply .cbmc-batch/jobs/aws_cryptosdk_string_dup/aws_cryptosdk_string_dup_harness.c:29]: Yes! There is actually such a string function but it's not usable as it starts with nondet_. There is a GitHub issue for it here. https://github.com/awslabs/aws-c-common/issues/718

File: .cbmc-batch/jobs/aws_cryptosdk_aes_gcm_decrypt/cbmc-batch.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tegansb) [reply .cbmc-batch/jobs/aws_cryptosdk_aes_gcm_decrypt/cbmc-batch.yaml:1]: Yes that is what I did in the previous #618!

[the remark this answers is not in the document]
Comment (tegansb) [reply .cbmc-batch/jobs/aws_cryptosdk_aes_gcm_decrypt/cbmc-batch.yaml:1]: I'll close #615 though

**Review by alex-chew — APPROVED** [review]

**Pull request merged**